src/Models/HydrostaticFreeSurfaceModels/SplitExplicitFreeSurfaces/split_explicit_timesteppers.jl

```diff
@@ -0,0 +1,154 @@
+
+Adapt.adapt_structure(to, t::AdamsBashforth3Scheme) = 
+    AdamsBashforth3Scheme(
+        Adapt.adapt(to, t.ηᵐ  ),
+        Adapt.adapt(to, t.ηᵐ⁻¹),
+        Adapt.adapt(to, t.ηᵐ⁻²),
+        Adapt.adapt(to, t.Uᵐ⁻¹),
+        Adapt.adapt(to, t.Uᵐ⁻²),
+        Adapt.adapt(to, t.Vᵐ⁻¹),
+        Adapt.adapt(to, t.Vᵐ⁻²),
+        t.β, t.α, t.θ, t.γ, t.δ, t.ϵ, t.μ)
+
+function materialize_timestepper(t::AdamsBashforth3Scheme, grid, free_surface, velocities, u_bc, v_bc)
+    ηᵐ   = free_surface_displacement_field(velocities, free_surface, grid)
+    ηᵐ⁻¹ = free_surface_displacement_field(velocities, free_surface, grid)
+    ηᵐ⁻² = free_surface_displacement_field(velocities, free_surface, grid)
+
+    Uᵐ⁻¹ = Field{Face, Center, Nothing}(grid; boundary_conditions = u_bc)
+    Uᵐ⁻² = Field{Face, Center, Nothing}(grid; boundary_conditions = u_bc)
+    Vᵐ⁻¹ = Field{Center, Face, Nothing}(grid; boundary_conditions = v_bc)
+    Vᵐ⁻² = Field{Center, Face, Nothing}(grid; boundary_conditions = v_bc)
+
+    FT = eltype(grid)
+
+    β = convert(FT, t.β)
+    α = convert(FT, t.α)
+    θ = convert(FT, t.θ)
+    γ = convert(FT, t.γ)
+    δ = convert(FT, t.δ)
+    ϵ = convert(FT, t.ϵ)
+    μ = convert(FT, t.μ)
+    
+    return AdamsBashforth3Scheme(ηᵐ, ηᵐ⁻¹, ηᵐ⁻², Uᵐ⁻¹, Uᵐ⁻², Vᵐ⁻¹, Vᵐ⁻², β, α, θ, γ, δ, ϵ, μ)
+end
+
+#####
+##### Timestepper extrapolations and utils
+#####
+
+function materialize_timestepper(name::Symbol, args...) 
+    fullname = Symbol(name, :Scheme)
+    TS = getglobal(@__MODULE__, fullname)
+    return materialize_timestepper(TS, args...)
+end
+
+initialize_free_surface_timestepper!(::ForwardBackwardScheme, args...) = nothing
+
+function initialize_free_surface_timestepper!(timestepper::AdamsBashforth3Scheme, η, velocities)
+    parent(timestepper.Uᵐ⁻¹) .= parent(velocities.U)
+    parent(timestepper.Vᵐ⁻¹) .= parent(velocities.V)
+
+    parent(timestepper.Uᵐ⁻²) .= parent(velocities.U)
+    parent(timestepper.Vᵐ⁻²) .= parent(velocities.V)
+
+    parent(timestepper.ηᵐ)   .= parent(η)
+    parent(timestepper.ηᵐ⁻¹) .= parent(η)
+    parent(timestepper.ηᵐ⁻²) .= parent(η)
+
+    return nothing
+end
+
+# The functions `η★` `U★` and `V★` represent the value of free surface, barotropic zonal and meridional velocity at time step m+1/2
+@inline U★(i, j, k, grid, t::ForwardBackwardScheme, Uᵐ) = @inbounds Uᵐ[i, j, k]
+@inline U★(i, j, k, grid, t::AdamsBashforth3Scheme, Uᵐ) = @inbounds t.α * Uᵐ[i, j, k] + t.θ * t.Uᵐ⁻¹[i, j, k] + t.β * t.Uᵐ⁻²[i, j, k]
+
+@inline η★(i, j, k, grid, t::ForwardBackwardScheme, ηᵐ⁺¹) = @inbounds ηᵐ⁺¹[i, j, k]
+@inline η★(i, j, k, grid, t::AdamsBashforth3Scheme, ηᵐ⁺¹) = @inbounds t.δ * ηᵐ⁺¹[i, j, k] + t.μ * t.ηᵐ[i, j, k] + t.γ * t.ηᵐ⁻¹[i, j, k] + t.ϵ * t.ηᵐ⁻²[i, j, k]
+
+@inline advance_previous_velocities!(::ForwardBackwardScheme,   i, j, k, U) = nothing
+@inline advance_previous_free_surface!(::ForwardBackwardScheme, i, j, k, η) = nothing
```

Comment:
I think other parts of the code call this something like "store_tendencies". That could be changed to "advance_tendencies"? Either way we should try not to create new terminology for existing concepts.